protocol/contracts/farm/facets/SiloFacet/DelegateFacet.sol

```diff
@@ -0,0 +1,71 @@
+/*
+ * SPDX-License-Identifier: MIT
+ */
+
+pragma solidity ^0.7.6;
+pragma experimental ABIEncoderV2;
+
+import "./Silo.sol";
+import "../../ReentrancyGuard.sol";
+import "../../../libraries/Silo/LibDelegate.sol";
+
+/*
+ * @author Publius
+ * @title DelegateFacet handles delegating authority for Beanstalk functions.
+ */
+contract DelegateFacet is Silo {
+    mapping(bytes4 => bool) public selectorEnabled;
+
+    bytes4 public constant PLANT_DELEGATED_SELECTOR =
+        bytes4(keccak256(bytes("plantDelegated(address)")));
+
+    constructor() {
+        selectorEnabled[PLANT_DELEGATED_SELECTOR] = true;
+    }
+
+    /**
+     * @notice delegateApproval sets approval value for delegation
+     * @param selector function selector
+     * @param delegatee contract/EOA address to delegate to
+     * @param approval approval value bytes32 of uint256 or bool
+     */
+    function delegateApproval(
+        bytes4 selector,
+        address delegatee,
+        bytes32 approval
+    ) external {
+        require(
+            selectorEnabled[selector],
+            "DelegateFacet: invalid function selector"
+        );
+        LibDelegate.setAllowance(msg.sender, selector, delegatee, approval);
+    }
+
```

Comment:
Can we add a `delegateAllowance` view function that returns the current allowance?